editor/src/clj/editor/collision_object.clj

```diff
@@ -545,6 +578,7 @@
                                                       :child-reqs [{:node-type Shape
                                                                     :tx-attach-fn (partial attach-shape-node true)}]}))
 
+  (output id-counts NameCounts :cached (g/fnk [shapes] (frequencies (mapv :id shapes))))
```

Comment:
Using `map` to produce a lazy sequence is fine here since `frequencies` consumes it.

The scary thing about lazy sequences is that the things they operate on cannot be garbage collected, and that you might get side effects such as exceptions or logging at an unexpected (for you) point in time. But we should feel free to use them to produce intermediate sequences that are consumed like this.